pkg/cache/cache_with_eviction.go

```diff
@@ -57,7 +57,10 @@ func NewLRUWithEviction[K comparable, V any](reg prometheus.Registerer, maxEntri
 
 // onEvicted is called when an entry is evicted from the underlying LRU.
 func (c *LRUWithEviction[K, V]) onEvicted(k K, v V) {
-	go c.onEvictedCallback(k, v)
+	c.mtx.Lock()
```

Comment:
`onEvicted` only gets called by one of the exported functions, and they are already locked. I believe this could try to lock the already locked lock.

Reply:
Good catch, thanks!